tmt/schemas/common.yaml

```diff
@@ -117,7 +117,23 @@ definitions:
   # https://tmt.readthedocs.io/en/stable/spec/tests.html#environment
   # https://tmt.readthedocs.io/en/stable/spec/plans.html#environment
   environment:
-    $ref: "/schemas/common#/definitions/object_with_string_properties"
+    type: object
+
+    additionalProperties:
+      anyOf:
+        - type: string
+        - type: boolean
+        - type: number
+        # Apparently, `number` is not enough to cover integers, according to
```

Comment:
As we don't care about `integer` versus `float` and allow both under `environment` values, wouldn't `type: number` be just enough? Seems to be working just fine.

Reply:
Hmm, it might be possible after all. I read the documentation WRT `integer`, which has some caveats, but `number` clearly says it covers both integers and floats. So yeah, `number` it is.

Reply:
Fixed.

Reply:
@happz

I am still confused, the schema validates the internal representation in the node data field, which is a string (tmt does conversion). Is this going to be changed? Otherwise, how could it be anything else?

Reply:
The schema validates the input data, not their internal reprezentation: input (raw fmf node) => validation (schema + python code) => normalization (python code) => valid & nice-to-use internal structures.

Validation is applied to input data in fmf nodes, then normalization steps in and, being aware of successful validation. Normalization is where `str(value)` is called for each item of `environment` - and after validation, normalization can be sure `str()` is a valid operation to take, because every item is either a string, a number, or boolean and nothing else. So, from possible `environment: false` we get to nice `Dict[str, str]` internally.